src/generators/modelsGenerator.ts

```diff
@@ -536,9 +536,7 @@ function writeUniontypes({ objects }: ClientDetails, modelsFile: SourceFile) {
     .forEach(obj => {
       const polymorphicObject = obj as PolymorphicObjectDetails;
       const childrenNames = [
-        ...(isPolymorphicParentInUnion(polymorphicObject)

```

Comment:
I don't think we have to change this

Reply:
We have to change this. I found this after our call. If you look into this file https://github.com/Azure/autorest.typescript/pull/804/files#diff-51f86c8fa85803f85a9039f559710dc4b574b28f7db2ca55e51743d26cd6dd37L509, the build will fail if you do not add the base class to the union.

Reply:
@joheredi without this change, you will see build errors such as:

```
Type 'Shark' is not assignable to type 'FishUnion'.
    Type 'Shark' is not assignable to type 'Cookiecuttershark'.
      Type 'Shark' is not assignable to type '{ fishtype: "cookiecuttershark"; }'.
        Types of property 'fishtype' are incompatible.
          Type '"shark"' is not assignable to type '"cookiecuttershark"'.
```

Reply:
Discussed Offline. Resolving